lib/main.dart

```diff
@@ -124,10 +127,59 @@ class MyApp extends StatefulWidget {
   State<MyApp> createState() => _MyAppState();
 }
 
-class _MyAppState extends State<MyApp> {
+class _MyAppState extends State<MyApp> with WidgetsBindingObserver {
   final secureStorage = SecureStorage();
   bool showOnboardingScreen = true;
 
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _checkTimeConsistency();
+    });
+  }
+
+  @override
+  void dispose() {
+    WidgetsBinding.instance.removeObserver(this);
+    super.dispose();
+  }
+
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) {
+    if (state == AppLifecycleState.resumed) {
+      _checkTimeConsistency();
+    }
+  }
+
+  Future<void> _checkTimeConsistency() async {
```

Comment:
@noahsilencelab Can you move this function in utils or somewhere else, this is making main.dart crowded.